Comparison Engine/Comparison Engine/Form1.cs

```diff
@@ -119,27 +132,34 @@ private void searchFunction(String keyword)
         //OPENING CHILD FORMS    #childforms
 
         public Form activeForm = null;
+        public MapForm mainMapForm = null;
         private void openChildFormMap(Form childForm)                           //this method will probably recieve some sort of mapData in the future
         {
             MapForm mapForm = new MapForm();
+            mainMapForm = mapForm;
             configureChildForm(mapForm);
         }
         private void openChildFormBar(Bar bar)
         {
-            BarForm barForm = new BarForm(bar);
-            //closeActiveForm();            //#commentedarea
+            BarForm barForm = new BarForm(bar);        
             configureChildForm(barForm);
         }
         private void openChildFormDrink(Drink drink)
         {
-            DrinkForm drinkForm = new DrinkForm(drink);
-            //closeActiveForm();            //#commentedarea
+            DrinkForm drinkForm = new DrinkForm(drink);            
             configureChildForm(drinkForm);
         }
+        private void openChildFormProfile()                 //This will probably recieve the user info
+        {
+            ProfileForm profileForm = new ProfileForm();         //This will probably recieve the user info
+            configureChildForm(profileForm);
+        }
         private void configureChildForm(Form childForm)
         {
-            //if (activeForm.GetType != MapForm.get)              //#commentedarea
-
+            if (activeForm != null && activeForm.GetType() != mainMapForm.GetType())
+            {
+                closeActiveForm();
+            }
```

Comment:
I think this could be a seperate method

Reply:
Why?

Reply:
maybe it makes sense, thought it breaks solid as seemed that it closes forms if it's not MapForm, but now realized that it closes the form that you're coming from only leaving the MapForm if a new form is initialized on top of it, so kinda makes sense